vllm/model_executor/models/mixtral.py

```diff
@@ -220,28 +220,36 @@ def process_weights_after_loading(self):
                                          requires_grad=False)
 
     def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
-        num_tokens, hidden_size = hidden_states.shape
+        batch_size, sequence_length, hidden_size = hidden_states.shape
         hidden_states = hidden_states.view(-1, self.hidden_size)
         # router_logits: (num_tokens, n_experts)
         router_logits, _ = self.gate(hidden_states)
-        final_hidden_states = fused_moe(hidden_states,
-                                        self.w13_weight,
-                                        self.w2_weight,
-                                        router_logits,
-                                        self.top_k,
-                                        renormalize=True,
-                                        inplace=True,
-                                        use_fp8=self.use_fp8,
-                                        w1_scale=self.w13_scale,
-                                        w2_scale=self.w2_scale,
-                                        a1_scale=self.a13_scale,
-                                        a2_scale=self.a2_scale)
+
+        if is_hpu():
+            final_hidden_states = static_fused_moe(hidden_states,
+                        self.w13_weight,
+                        self.w2_weight,
+                        router_logits,
+                        self.top_k)
+        else:
+            final_hidden_states = fused_moe(hidden_states,
+                                            self.w13_weight,
+                                            self.w2_weight,
+                                            router_logits,
+                                            self.top_k,
+                                            renormalize=True,
+                                            inplace=True,
+                                            use_fp8=self.use_fp8,
+                                            w1_scale=self.w13_scale,
+                                            w2_scale=self.w2_scale,
+                                            a1_scale=self.a13_scale,
+                                            a2_scale=self.a2_scale)
 
         if self.tp_size > 1:
             final_hidden_states = tensor_model_parallel_all_reduce(
                 final_hidden_states)
 
-        return final_hidden_states.view(num_tokens, hidden_size)
+        return final_hidden_states.view(batch_size, sequence_length, hidden_size)
```

Comment:
this is HPU specific and will break on other accelerators. Please cover that with is_hpu.

Reply:
Done